src/main/java/org/elasticsearch/river/wikipedia/support/WikiXMLParser.java

```diff
@@ -87,6 +90,11 @@ protected InputSource getInputSource() throws Exception {
 
     protected void notifyPage(WikiPage page) {
         currentPage = page;
+    }
 
+    public void close() throws IOException {
+        if (cbZip2InputStream != null) {
+            cbZip2InputStream.close();
```

Comment:
I wonder why the underlying InputStream must be explicitly closed? Closing the BufferedReader should propagate the close, no?